prow/spyglass/spyglasstests/spyglass_test.go

```diff
@@ -0,0 +1,424 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//Package spyglasstests contains tests for spyglass
+package spyglasstests
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"github.com/fsouza/fake-gcs-server/fakestorage"
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/deck/jobs"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/spyglass"
+)
+
+var (
+	fakeGCSBucket    *storage.BucketHandle
+	testAf           *spyglass.GCSArtifactFetcher
+	fakeJa           *jobs.JobAgent
+	fakeGCSJobSource *spyglass.GCSJobSource
+	buildLogName     string
+	longLogName      string
+	junitName        string
+	startedName      string
+	finishedName     string
+	junitKey         string
+	buildLogKey      string
+	longLogKey       string
+	startedKey       string
+	finishedKey      string
+)
+
+const (
+	testSrc = "gs://test-bucket/logs/example-ci-run/403"
+)
+
+type fkc []kube.ProwJob
+
+func (f fkc) GetLog(pod string) ([]byte, error) {
+	return nil, nil
+}
+
+func (f fkc) ListPods(selector string) ([]kube.Pod, error) {
+	return nil, nil
+}
+
+func (f fkc) ListProwJobs(s string) ([]kube.ProwJob, error) {
+	return f, nil
+}
+
+type fpkc string
+
+func (f fpkc) GetLog(pod string) ([]byte, error) {
+	if pod == "wowowow" || pod == "powowow" {
+		return []byte(f), nil
+	}
+	return nil, fmt.Errorf("pod not found: %s", pod)
+}
+
+func (f fpkc) GetContainerLog(pod, container string) ([]byte, error) {
+	if pod == "wowowow" || pod == "powowow" {
+		return []byte(f), nil
+	}
+	return nil, fmt.Errorf("pod not found: %s", pod)
+}
+
+func TestMain(m *testing.M) {
+	fakeGCSJobSource = spyglass.NewGCSJobSource(testSrc)
+	testBucketName := fakeGCSJobSource.BucketName()
+	buildLogName = "build-log.txt"
```

Comment:
comments on shared globals in tests are outstanding

Reply:
fixed in latest